server.cc

```diff
@@ -42,19 +42,28 @@ using namespace ::apache::thrift::server;
 using std::shared_ptr;
 
 struct fault_descriptor {
-    bool random;        // error code must be randomized
-    int err_no;         // error code to return
-    int32_t probability;    // 0 < probability < 100
-    std::string regexp; // regular expression on filename
-    bool kill_caller;   // Must we kill the caller
-    int32_t delay_us;       // operation delay in us
-    bool auto_delay;    // must auto delay like an SSD
+    bool random;                                // error code must be randomized
+    std::vector<int32_t>  err_nos;              // error code(s) to select from
+    int32_t probability;                        // 0 < probability < 100
+    std::string regexp;                         // regular expression on filename
+    bool kill_caller;                           // Must we kill the caller
+    int32_t delay_us;                           // operation delay in us
+    bool auto_delay;                            // must auto delay like an SSD
 };
 
+std::vector<int32_t> default_errnos;
 std::set<std::string> valid_methods;
 std::map<std::string, fault_descriptor> fault_map;
 std::mutex mutex;
 
+void init_default_errnos()
+{
+    int32_t errno;
+    for (errno = E2BIG; errno < EXFULL; errno++) {
```

Comment:
I see where this range comes from, but it doesn't make a lot of sense to be honest.
As a follow up we should be more specific about the random errors and drop those that don't make sense, like `ENOSYS` or `EL2NSYNC` and OTOH add some that do make sense like `EIO`.